h/services/group.py

```diff
@@ -120,6 +120,20 @@ def groupids_created_by(self, user):
 
         return [g.pubid for g in self.session.query(Group.pubid).filter_by(creator=user)]
 
+    def _create(self, name, userid, description, joinable_by, readable_by, writeable_by):
+        """Create a group and save it to the DB."""
+        creator = self.user_fetcher(userid)
```

Comment:
I have a hunch there's a solid historical reason for this, but why do these service methods take a `userid` only (and then have to use the user fetcher service)? Is it possible to pass the entire `user` object around or is that bad/impracticable ?

Reply:
Honestly I'm not sure. It does kind of seem to me that they should take an actual user object. Then the view (private group create form and open group create CLI) would be responsible for finding the user based on the userid and giving an error message if the user doesn't exist. Currently there's no facility here for error handling if the user doesn't exist. It only really applies to the CLI currently since the private group create form uses the authenticated user as the creator. So I think we may want to refactor this in future